src/game/gamecore.h

```diff
@@ -236,8 +236,8 @@ class CCharacterCore
 	vec2 m_HookTeleBase;
 	int m_HookTick;
 	int m_HookState;
-	int m_HookedPlayer;
 	std::set<int> m_AttachedPlayers;
+	int GetHookedPlayer() const { return m_HookedPlayer; }
```

Comment:
```suggestion
	int HookedPlayer() const { return m_HookedPlayer; }
```
I think our getters usually don't include the word `Get`.

Reply:
> our getters usually don't include the word `Get`.

Let's see:
```cpp
CEntity::GetPos()
CEntity::GetProximityRadius()
CEntity::GetOwnerID()

CEntity::GameWorld()
CEntity::Config()
CEntity::GameServer()
CEntity::Server()
CEntity::Collision()

CPlayer::GetTeam()
CPlayer::GetCID()
CPlayer::GetUniqueCID()
CPlayer::GetClientVersion()
CPlayer::GetCharacter()

CPlayer::GameServer()
CPlayer::Server()

IServer::Tick()
IServer::TickSpeed()

IServer::Port()
IServer::MaxClients()
IServer::ClientCount()
IServer::DistinctClientCount()
IServer::ClientName()
IServer::ClientClan()
IServer::ClientCountry()
IServer::ClientIngame()
IServer::ClientAuthed()
IServer::GetClientInfo()
IServer::GetClientAddr()
IServer::GetMapInfo()
IServer::GetAuthedState()
IServer::GetAuthName()
IServer::GetClientAddr()
IServer::GetIdMap()
IServer::GetAnnouncementLine()
IServer::GetNetErrorString()
IServer::GetMapName()

IMap::GetData()
IMap::GetDataSize()

IMap::GetItem()
IMap::GetItemSize()
IMap::GetType()
IMap::NumItems()

IEngineMap::Sha256()
IEngineMap::MapSize()
IEngineMap::Crc()
```

I'd say that it is super confusion to have:
```
CServer::Tick()
CGameContext::OnTick()
CGameWorld::Tick()
CEntity::Tick()
IGameController::Tick()
CPlayer::Tick()
```
`CGameContext::OnTick()` is called by the server to process the game state. GameContext calls `Tick()` on many other objects but `CServer::Tick() const` returns the current game tick, and all other (`CGameWorld::Tick()`) methods process one game tick (move the entities, etc).
It would be much better if `CServer::Tick()` had name `CServer::GetTick()`. Or those few dozens `<class>::Tick()` could be `OnTick()` (and then we'd also need `OnSnap()`, etc for consistency.

The client side CGameWorld has `GameTick()` and `GameTickSpeed()`. It is much better than `CServer::Tick()` and I tried to do the same on the server side (but we still need Server() in entities, so that change alone does not allow us to reduce dependencies, and the MR is already big so I hold it in local WIP commit).

Reply:
I agree that the `Tick` function naming is unfortunate. I'd guess it comes from the fact that `Tick` is both a verb and a noun. This also affects `Snap`, but that doesn't seem to be used as a function name.

Generally, I'd favor shorter names, "get" doesn't seem to add much to the meaning of the functions.

Reply:
> This also affects Snap, but that doesn't seem to be used as a function name.

Probably you mean that Snap is used *only* as a (member) function name:

> https://github.com/ddnet/ddnet/blob/3917e191e9c865443708cb2753ded8491eb11359/src/game/server/entity.h#L119C7-L119C7
> `void CEntity::Snap(int SnappingClient) {}`

I have a strong impression that Teeworlds code convention is to use that `Get` prefix and the most of getters have it. DDNet has a long history and there was a lot of code style violations, and I counted missing `Get` prefix as one of those.

So the long-term direction is rename
- CGameWorld::Tick()
- CEntity::Tick()
- IGameController::Tick()
- CPlayer::Tick()
- ...

because those sound like getters, and then also drop `Get` prefix from those
- CEntity::GetPos()
- CEntity::GetProximityRadius()
- CPlayer::GetTeam()
- CPlayer::GetCID()
- CPlayer::GetClientVersion()
- CPlayer::GetCharacter()
- ...

for consistency. Okay, if that is the DDNet direction. 👀